packages/core/src/RenderingEngine/BaseVolumeViewport.ts

```diff
@@ -280,23 +280,39 @@ abstract class BaseVolumeViewport extends Viewport implements IVolumeViewport {
       return;
     }
     const { volumeActor } = applicableVolumeActorInfo;
+    const vtkImageData = volumeActor.getMapper().getInputData();
+    const interval = vtkImageData.getSpacing()[2];
+    /*
+     Opacity of a volume is nonlinear. Specifically in volume rendering opacity is interpreted in terms of opacity per unit distance.
+     When you need to take an interval of a volume (say 3mm of a ray) and convert that to opacity you use a power function to accumulate the opacity over the interval.
+     For example if your opacity is 0.5/mm and the interval is 3mm long then the opacity is 1.0 - pow(1.0 - 0.5, 3) or 0.875
+     https://github.com/Kitware/vtk-js/pull/2093
+     */
+    const convertOpacityToVTKOPacity = (opacity: number, interval: number) => {
```

Comment:
This const doesn't depend on any local values, so please move up to a top level function so it doesn't get redefined